### PR TITLE
Modifying code examples to include training data.

### PR DESCRIPTION
**Pull request by toshihikoyanase**
## Motivation
This PR resolves https://github.com/optuna/optuna/issues/1049. I'll take over @barneyhill's work in https://github.com/optuna/optuna/pull/1051 because he is not available due to his machine trouble.

## Description of the changes

- Replace randomly generated datasets with the iris dataset.
- The Hyperband example introduced in #972 is also updated.

### CODE REVIEW

**Comment by hvy** [comment]
Oh seems like the trial refactoring lead to a conflict. Could you fix it?

**Comment by hvy** [comment]
I guess we should wait for https://github.com/optuna/optuna/pull/1167.

**Comment by hvy** [comment]
#1167 is now merged. Quite a few conflicts but I guess we can continue with this one.

File: optuna/pruners/hyperband.py [hunk 1]
```diff
@@ -63,22 +63,17 @@ class HyperbandPruner(BasePruner):
 
         We minimize an objective function with Hyperband pruning algorithm.
 
-        .. testsetup::
+        .. testcode::
 
-            import numpy as np
+            from sklearn.datasets import load_iris
```
Comment (hvy) [inline optuna/pruners/hyperband.py:68]: Was the `numpy` import intentionally removed? It still seems to pass `doctest` so I guess it's imported in some preamble but it seemed a bit inconsistent to me.

File: optuna/trial/_trial.py [hunk 2]
```diff
@@ -96,20 +96,18 @@ def suggest_float(self, name, low, high, *, log=False, step=None):
             Suggest a momentum, learning rate and scaling factor of learning rate
             for neural network training.
 
-            .. testsetup::
+
```
Comment (hvy) [inline optuna/trial/_trial.py:99]: Might not need this extra blank line.
```suggestion
```

Comment (toshihikoyanase) [reply optuna/trial/_trial.py]: Thanks! I applied your suggestion in https://github.com/optuna/optuna/pull/1221/commits/55a446cbb955effa6103e826e98022f6fb335d90.

File: optuna/pruners/hyperband.py (hunk 1 above)
Comment (toshihikoyanase) [reply optuna/pruners/hyperband.py:69]: You're absolutely right. I happened to remove it during git-rebase. I add `numpy` in https://github.com/optuna/optuna/pull/1221/commits/825a207250dc933e77738b7e07e582ddf81ea140. Thank you for your careful review.

**Pull request merged**